crates/kstring/src/cow.rs

```diff
@@ -75,6 +75,12 @@ impl<'s> KStringCow<'s> {
         self.inner.as_str()
     }
 
+    /// Returns a string slice, only if this is borrowed
+    #[inline]
+    pub fn as_borrowed(&self) -> Option<&'s str> {
```

Comment:
The reason the underlying variants are not exposed is that I didn't feel comfortable getting locked down to a specific implementation.  `KStringCow` effectively deals with the following states:
- `'static`
- `'s`
- Heap-string
- Stack-string

I switched around how I dealt with these several times during the creation of `kstring`.

I feel like an `as_borrowed` doesn't accurately map as-is t-1 with what `KStringCow` supports because it treats `'static` as owned. I also feel a little dirty with us not having symmetric cases for owned.

What if instead we had a `to_str_cow` that
- `'static` and '`s' map to `str`
- Heap/Stack strings map to `String`

Reply:
Ah, I had missed the inners of `KString`, didn't realize `'static` was hidden in there.